### PR TITLE
refactoring: Use `tryJust` instead of pattern guards

### PR DESCRIPTION
**Pull request by sol**
Doing this first will give us a smaller diff for #70, hopefully putting us into a better position to make progress there.

This reduces the diff for #70 to essentially: https://github.com/sol/typed-process/commit/4c5d917b740d4258865027ddff49f2848780edfe

But ignoring #70 for a moment, I still think this is a reasonable improvement.

### CODE REVIEW

**Comment by tomjaguarpaw** [comment]
I found this a little hard to follow.  Could you please rebase on https://github.com/tomjaguarpaw/typed-process/tree/terminateProcess for clarity? That makes each of the steps painfully obvious.

**Comment by sol** [comment]
> I found this a little hard to follow. Could you please rebase on https://github.com/tomjaguarpaw/typed-process/tree/terminateProcess for clarity? That makes each of the steps painfully obvious.

done.

**Comment by tomjaguarpaw** [comment]
If find the `tryJust` and `guard` much harder to understand, actually, but I think my three commits are equally good for your purpose, @sol, are they not? Unless they're not, I intend to merge them in https://github.com/fpco/typed-process/pull/72.

**Comment by sol** [comment]
Anything works for me.  Closing in favor of https://github.com/fpco/typed-process/pull/72.
